### PR TITLE
Move actions + types from /native to /core

### PR DESCRIPTION
**Pull request by LeoW93**
- We need to have the actions from `native` available in the agency hub
- This PR moves the actions and types over to `core` so they can be used by both native and web applications
- Also adds the `skipCapture` and `skipDisplay` options into the `reportError` action creator options argument so we don't need to `// @ts-ignore` everywhere 🙃 

Also makes use of a [`yarn` workspace ](https://yarnpkg.com/lang/en/docs/workspaces/) structure so that the code written in `core` can be immediately used in the `native` library (for local dev, CI etc)

### CODE REVIEW

File: core/src/hydra/redux.ts [hunk 1]
```diff
@@ -2,10 +2,11 @@ import {
   reduceList as originalReduceList,
   reduceItems as originalReduceItems,
   reduceListAndItems as originalReduceListAndItems,
-  Action,
   ReduceListOptions, ActionLifecycleOptions,
 } from "../store/redux/reducers";
 
+import {Action} from "redux";
```
Comment (sroze) [inline core/src/hydra/redux.ts]: Could you add `prettier`?

File: native/src/errors/index.ts [hunk 2]
```diff
@@ -19,7 +18,6 @@ export {
   components,
   middleware,
   reducer,
-  actions,
```
Comment (sroze) [inline native/src/errors/index.ts]: For BC, I suggest you still export them (import them from core).

File: core/src/store/redux/actions.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { ReportActionOptions, ReportErrorActionCreatorWithType, DismissErrorActionCreatorWithType } from "./types";
+import typedActionCreatorFactory from "../../typed-action-creator/factory";
+
+const randomIdentifier = () => Math.random().toString(36).substring(2, 15)
```
Comment (fenderface66) [inline core/src/store/redux/actions.ts]: Out of interest why do we not use `uuid` here?

Comment (LeoW93) [reply core/src/store/redux/actions.ts]: not sure - one less dependency? 🤷‍♂
one for @sroze I would say..

Comment (sroze) [reply core/src/store/redux/actions.ts]: I don't remember tbh.

**Pull request merged**